### ISSUE TITLE
Atom Helper high CPU usage

### ISSUE THREAD

**Issue by LukeMauldin** [atom/atom#3426]
Since upgrading to 0.123 (or maybe 0.124) I have been experiencing very high CPU usage from Atom Helper.  Running on Macbook Pro with OSX 10.9.4.  Have been using Atom for several months and have not experienced this problem before.  Unsure how to troubleshoot more.

**Comment by wmertens** [comment]
I'm seeing this with 0.179.0 still. atom-helper "not responding" and 20-33% CPU in the Activity Monitor while Atom is idling.

Memory usage:

Here's a sampling of it: https://gist.github.com/wmertens/56aae47cab9eed0b1027

**Comment by fbender** [comment]
I experience this issue as well and disabled all community packages (except lang-matlab-octave) and some core packages in v0.180.0 and *still* see an almost constant CPU usage of 5-8% with Atom in the background (window is not visible behind other app windows) and having 5 small to medium large HTML, CSS, and JS files open. Energy Factor is ~7 on my MBA 2012 with OS X 10.9.5. Having the window partially visible (but still in background) I can scroll a file which kicks the CPU up to 50-100% and Energy Factor to no less than 90.

**Comment by zsgalusz** [comment]
I disabled every package including the default ones (save for settings-view) and AtomHelper was still gnawing 1-2% while idling in the background. Just switching to Atom and scrolling the settings spawns another AtomHelper using 30-50% CPU that stays until I switch away again. This is on a 2013 rMBP, Atom 0.182.0.

**Comment by mathie** [comment]
Yeah, this matches what I'm seeing. The background CPU usage seems to be proportional to the number of open documents in a project. Right now, with 2 open documents, it's hovering at around 3%. With 18 open documents, it has settled down to ~20% background CPU usage. Activity monitor is showing the relative impact on the battery of keeping Atom open:

![activity_monitor__applications_in_last_8_hours__and_atom_helper_high_cpu_usage_ _issue__3426_ _atom_atom_and_my-favourite-interview-question_markdown_-__users_mathie_development_personal_mathie_github_io_-_atom](https://cloud.githubusercontent.com/assets/438/6387066/7cd7f9be-bd7c-11e4-95ad-3773b9cc3522.png)

It's making Safari look good. ;-)

The helper seems to be the culprit:

![atom_helper__61502_](https://cloud.githubusercontent.com/assets/438/6387079/b91fa2b4-bd7c-11e4-8163-5dbe4196861e.png)

Here's a random sample from while I was updating this issue: https://gist.github.com/mathie/7cca66994db9bcaefc1f

I'm on Mac OS X 10.10.3, Atom 0.182.0, on a 2013 Macbook Pro. All the files I had open at the time were Markdown text (if that's significant, plugin-wise). If there's any more information I can provide, please do ask.

**Comment by mathie** [comment]
Running Atom in safe mode, the background CPU usage is less (~10% on average, I'd say). Here's a sample from safe mode: https://gist.github.com/mathie/d1742725372493b3af9a

**Comment by maxogden** [comment]
I am also experiencing this issue, it is causing a noticable reduction in the amount of time my laptop battery lasts

![screen shot 2015-03-05 at 2 57 22 pm](https://cloud.githubusercontent.com/assets/39759/6516595/2431d2f4-c348-11e4-95c0-78c36be46171.png)

**Comment by Antikythera** [comment]
Hello guys, I just noticed this problem on 2015 MacbookAir. I was shocked to see that with 60% battery I had estimated 4-5 hours. So ofc I came to a same conclusion as you after seeing the results in activity monitor.

**Comment by Zireael07** [comment]
What are your linters set to? On save, on focus?

**Comment by zsgalusz** [comment]
I disabled everything, including linters.

**Comment by Antikythera** [comment]
I wasn't even aware I could set that. Where do I look for those settings? It's not on the settings page.

**Comment by Zireael07** [comment]
@Antikythera: Click on the settings button under linter and you have lint on change, lint on editor focus, lint on save options.

**Comment by Antikythera** [comment]
@Zireael07 Ok I had to manually change my config file, after reloading those options showed up in the settings tab for linter. I will play around with these and post what I found out.

**Comment by Antikythera** [comment]
So after a couple of hours of usage, and some linter config tweaks, this is the result.
![screen shot 2015-03-31 at 7 00 03 pm](https://cloud.githubusercontent.com/assets/5049882/6924336/2f9b52ba-d7d8-11e4-8f2c-11fe1ef74ea0.png)
So first thing I had to do to enable the settings for linter is edit a empty section for lint in config file.
```coffee
linter:
    LintOnSave: true
    LintOnChange: true
    lintOnChangeInterval: 5000
```
It is only after I did this that I could see all the settings for linter. The whole thing was just missing.
So these are my current settings for linter.
![screen shot 2015-03-31 at 7 06 58 pm](https://cloud.githubusercontent.com/assets/5049882/6924491/2a0b5466-d7d9-11e4-95ce-521f19c6b2cd.png)
5 seconds for linting may be a bit sluggish but the impact on battery life is huge, I will try lowering this and see how low I can get it and still keep a sane power usage.

**Comment by Zireael07** [comment]
So, 314 impact is w/o linter and 718 w/ linter?

That points to linters possibly being to blame, not atom core.

**Comment by Antikythera** [comment]
That average is taken from last 8 hours, so it's not really a good indicator since I didn't use atom equally when these screenshots were taken. But linters **definitely** have something to do with this issue, and the fact that increasing the delay between linting dramatically reduces energy impact just proves it.
I would really like if somebody could confirm this before we dismiss this problem tho.

**Comment by jtwebman** [comment]
It seems to me that the linters should look at Reactive Ex libraries. I have never built a Atom plugin but instead of pulling ever second or 5 seconds they should just catch the OnTestChange event coming from the editor control. Maybe there isn't one and that is the issue.

**Comment by DavidAwad** [comment]
+1, I've got 4 windows open on the highest spec'ed 2015 15 inch pro, and it's using 53% and 28% of my cpu in two separate processes.

**Comment by thomasjo** [comment]
@DavidAwad Can you reproduce what you're seeing when running Atom in safe mode? Completely shutdown all Atom instances and launch safe mode via `atom --safe`. I'm 99% sure your issue is caused by bad packages.

**Comment by mdalpi** [comment]
I don't know if this helps but with a clean installation, safe booting and linters disabled I'm still getting high CPU usage when scrolling and it gets worse in larger Python files (>100%).

Here's a video (sorry for the host, don't use YouTube myself)
https://sendvid.com/887rxkjt

And here's the Javascript CPU Profile:
https://gist.github.com/mdalpi/a14f8e9da2e9a32f7b3b

OS X 10.10.3
(Old) Core 2 Duo

**Comment by f6v** [comment]
Same here on Atom 1.0.0, OS X 10.10.13, MBP Retina 15 mid 2014

**Comment by cage1016** [comment]
Same problem on Atom 1.0.0, OS X 10.10.3, MBP Retina 15 mid 2015
Even run atom --safe got the same Atom Helper high CPU result.

**Comment by yibuyisheng** [comment]
@cage1016  +1

**Comment by methodbox** [comment]
I've seen the same. I've been following Atom and just heard it hit 1.0 and decided to check it out to see if I can mod it to provide everything Sublime does for me in a better way.

I had not had it open more than 30 seconds before I noticed my CPU was at 60 degrees and the fans were humming. No other apps were open.

I shut it down, waited for the temp to drop to normal (about 39 Celsius) and CPU usage to drop to 0, opened it again and the reaction was instant.

I will run tests like those above when I get home but I wanted to post because I found that similar issues occurred in Chrome and this issue strongly reminds me of the same thing.
I ultimately isolated the issue to how Chrome uses its rendering processes and the amount of memory allowed for each, as well as whether it has access to the discreet GPU.
Based solely on the fact that Atom appears to increase both memory usage and CPU usage when doing apparently nothing, my gut tells me this issue is related to the rendering engine and how it's allowed to offload processes to the GPU.
These can be modified with Chrome flags. Anyone know how to do this in Atom?

**Comment by truevektordev** [comment]
I'm seeing over 200% in activity monitor. If quit the process using activity monitor, everything seems to operate without issue...
![screen shot 2015-07-08 at 11 52 38 am](https://cloud.githubusercontent.com/assets/5365433/8576466/f82f9862-2567-11e5-894d-f2c36c922d42.png)

**Comment by methodbox** [comment]
Okay, so I said I'd run the tests above and see what results were achieved.... however, I didn't get much further than simply opening a Meteor project folder for a relatively small app before my CPU usage shot up to 80% and the temps hit 60C. On open, Atom seemed to barely even be noticed by the CPU; loading a project folder boosted CPU usage instantly and it stayed high.

It's pretty clear that since importing a project folder results in simply reading the file and displaying the folders and files in the tree this issue is triggered by some kind of indexing, so I started removing and importing different project folders, creating new empty folders, etc. It didn't seem to trigger any kind of change.

I noticed in notes above that users had some success removing their ```.git``` folder if it existed in the ~home directory.  This seemed to do the trick for me but I wanted to know WHY.

I took a process snapshot prior to doing this, and one after.  I don't know what these two blocks of code do, but there's two particular items that do NOT appear in the active processes after removing the ```.git``` directory:  

```
1951 Thread_20664: WorkerPool/30231
    + 1951 thread_start  (in libsystem_pthread.dylib) + 13  [0x7fff895d641d]
    +   1951 _pthread_start  (in libsystem_pthread.dylib) + 176  [0x7fff895d81e5]
    +     1951 _pthread_body  (in libsystem_pthread.dylib) + 131  [0x7fff895d8268]
    +       1951 base::PlatformThread::Join(base::PlatformThreadHandle)  (in libchromiumcontent.dylib) + 283  [0x10c47b19b]
    +         1951 base::PosixDynamicThreadPool::WaitForTask()  (in libchromiumcontent.dylib) + 1574  [0x10c4840c6]
    +           1951 base::PosixDynamicThreadPool::WaitForTask()  (in libchromiumcontent.dylib) + 188  [0x10c483b5c]
    +             1951 base::ConditionVariable::TimedWait(base::TimeDelta const&)  (in libchromiumcontent.dylib) + 91  [0x10c47517b]
    +               1951 __psynch_cvwait  (in libsystem_kernel.dylib) + 10  [0x7fff90866136]
    1951 Thread_20665: WorkerPool/30475
      1951 thread_start  (in libsystem_pthread.dylib) + 13  [0x7fff895d641d]
        1951 _pthread_start  (in libsystem_pthread.dylib) + 176  [0x7fff895d81e5]
          1951 _pthread_body  (in libsystem_pthread.dylib) + 131  [0x7fff895d8268]
            1951 base::PlatformThread::Join(base::PlatformThreadHandle)  (in libchromiumcontent.dylib) + 283  [0x10c47b19b]
              1951 base::PosixDynamicThreadPool::WaitForTask()  (in libchromiumcontent.dylib) + 1574  [0x10c4840c6]
                1951 base::PosixDynamicThreadPool::WaitForTask()  (in libchromiumcontent.dylib) + 188  [0x10c483b5c]
                  1951 base::ConditionVariable::TimedWait(base::TimeDelta const&)  (in libchromiumcontent.dylib) + 91  [0x10c47517b]
                    1951 __psynch_cvwait  (in libsystem_kernel.dylib) + 10  [0x7fff90866136]
```

The ```WaitForTask()``` stood out to me so I did a search in the snapshot file from after removing the git dir.  Neither the ```WaitForTask()``` function is anywhere to be found, nor is a thread called ```WorkerPool```.

I then ran a new ```git init``` in my home directory and restarted Atom.  The CPU jumped to 100% as soon as I opened a file. I took another snapshot and again, the ```WorkerPool``` and ```WaitForTask()``` culprits reappeared.

Removing the git dir again resolved the issue.  It would appear there is an issue with monitoring file changes in conjunction with creating commits from within Atom, but that's just a guess.

**Comment by methodbox** [comment]
Actually, I think I discovered the ```.git``` in ```home``` on this thread:
https://discuss.atom.io/t/atomhelper-high-cpu-usage/2553/46

So, I followed the steps of user "chrisb" and removed the directory.  But I also wanted to see if the default repo path changed with a git dir in the home folder vs. without:

With:
```
atom.project.getRepositories()
[GitRepositoryemitter: EmitterhandlerPath: "/Applications/Atom.app/Contents/Resources/app.asar/src/repository-status-handler.js"path: "/Users/USERNAME/.git"project: ObjectprojectAtRoot: falserepo: RepositorystatusTask: Taskstatuses: Objectsubscriptions: CompositeDisposableupstream: Object__proto__: GitRepository
```

Without:
```
[null]
```

Why is Atom defaulting to a repo directory I never specified?

**Comment by methodbox** [comment]
#4019, #1827,  #6437 are probably duplicates of this same issue.

**Comment by methodbox** [comment]
So I'm assuming since no one has responded since I posted my results it magically fixed this for everyone so you've stopped contributing?

**Comment by truevektordev** [comment]
Nope, it's still hammering my CPU.

**Comment by truevektordev** [comment]
I'm not super familiar with debugging desktop applications in OSX, but I'm thinking the problem might be here:

v8::Testing::DeoptimizeAll()  (in libchromiumcontent.dylib) + 0  [0x105ea84b0]

I took a sample from the offending process in found this:
```
Total number in stack (recursive counted multiple, when >=5):
        395       v8::Testing::DeoptimizeAll()  (in libchromiumcontent.dylib) + 0  [0x105ea84b0]
        83       ???  (in <unknown binary>)  [0x575979071bb]
        40       ???  (in <unknown binary>)  [0x575979068f5]
        22       v8::Testing::DeoptimizeAll()  (in libchromiumcontent.dylib) + 1499209  [0x1060164f9]
        22       v8::Testing::DeoptimizeAll()  (in libchromiumcontent.dylib) + 1500455  [0x1060169d7]
        22       v8::Testing::DeoptimizeAll()  (in libchromiumcontent.dylib) + 1506757  [0x106018275]
        22       v8::Testing::DeoptimizeAll()  (in libchromiumcontent.dylib) + 1507053  [0x10601839d]
        22       v8::Testing::DeoptimizeAll()  (in libchromiumcontent.dylib) + 1507101  [0x1060183cd]
        22       v8::Testing::DeoptimizeAll()  (in libchromiumcontent.dylib) + 1835747  [0x106068793]
        22       v8::Testing::DeoptimizeAll()  (in libchromiumcontent.dylib) + 1839796  [0x106069764]
        21       v8::Testing::DeoptimizeAll()  (in libchromiumcontent.dylib) + 1380257  [0x105ff9451]
        20       v8::Testing::DeoptimizeAll()  (in libchromiumcontent.dylib) + 1509960  [0x106018ef8]
        13       ???  (in <unknown binary>)  [0x57597a2f2ab]
        12       __psynch_mutexwait  (in libsystem_kernel.dylib) + 0  [0x7fff887fc15c]
        11       _platform_memmove$VARIANT$Unknown  (in libsystem_platform.dylib) + 0  [0x7fff80643fa0]
        11       v8::Testing::DeoptimizeAll()  (in libchromiumcontent.dylib) + 1510101  [0x106018f85]
        11       v8::Testing::DeoptimizeAll()  (in libchromiumcontent.dylib) + 3481662  [0x1061fa4ee]
        10       v8::Testing::DeoptimizeAll()  (in libchromiumcontent.dylib) + 1256597  [0x105fdb145]
        10       v8::Testing::DeoptimizeAll()  (in libchromiumcontent.dylib) + 1290687  [0x105fe366f]
        9       ???  (in <unknown binary>)  [0x575979062d5]
        9       ???  (in <unknown binary>)  [0x57597a67d0d]
        9       uv_mutex_lock  (in Atom Framework) + 9  [0x1041fdec4]
        8       ???  (in <unknown binary>)  [0x5759799e4f3]
        8       ???  (in <unknown binary>)  [0x5759799e67b]
        8       ???  (in <unknown binary>)  [0x57597a6822b]
        8       _pthread_body  (in libsystem_pthread.dylib) + 131  [0x7fff87a8b268]
        8       _pthread_start  (in libsystem_pthread.dylib) + 176  [0x7fff87a8b1e5]
        8       thread_start  (in libsystem_pthread.dylib) + 13  [0x7fff87a8941d]
        8       v8::Testing::DeoptimizeAll()  (in libchromiumcontent.dylib) + 1511574  [0x106019546]
        8       v8::Testing::DeoptimizeAll()  (in libchromiumcontent.dylib) + 3420579  [0x1061eb653]
        8       v8::Testing::DeoptimizeAll()  (in libchromiumcontent.dylib) + 3420760  [0x1061eb708]
        7       ???  (in <unknown binary>)  [0x5759799e74d]
        7       uv_mutex_unlock  (in Atom Framework) + 9  [0x1041fdf01]
        7       v8::Testing::DeoptimizeAll()  (in libchromiumcontent.dylib) + 1177505  [0x105fc7c51]
        7       v8::Testing::DeoptimizeAll()  (in libchromiumcontent.dylib) + 1251598  [0x105fd9dbe]
        7       v8::Testing::DeoptimizeAll()  (in libchromiumcontent.dylib) + 3193852  [0x1061b40ac]
        7       v8::Testing::DeoptimizeAll()  (in libchromiumcontent.dylib) + 3334174  [0x1061d64ce]
        7       v8::Testing::DeoptimizeAll()  (in libchromiumcontent.dylib) + 3421126  [0x1061eb876]
        6       OSAtomicCompareAndSwapPtrBarrier$VARIANT$mp  (in libsystem_platform.dylib) + 0  [0x7fff80642340]
        6       __psynch_cvwait  (in libsystem_kernel.dylib) + 0  [0x7fff887fc12c]
        6       __psynch_mutexdrop  (in libsystem_kernel.dylib) + 0  [0x7fff887fc144]
        6       v8::Testing::DeoptimizeAll()  (in libchromiumcontent.dylib) + 1254969  [0x105fdaae9]
        6       v8::Testing::DeoptimizeAll()  (in libchromiumcontent.dylib) + 1265672  [0x105fdd4b8]
        6       v8::Testing::DeoptimizeAll()  (in libchromiumcontent.dylib) + 1290663  [0x105fe3657]
        6       v8::Testing::DeoptimizeAll()  (in libchromiumcontent.dylib) + 1505259  [0x106017c9b]
        6       v8::Testing::DeoptimizeAll()  (in libchromiumcontent.dylib) + 1505394  [0x106017d22]
        6       v8::Testing::DeoptimizeAll()  (in libchromiumcontent.dylib) + 1545502  [0x1060219ce]
        6       v8::Testing::DeoptimizeAll()  (in libchromiumcontent.dylib) + 1878665  [0x106072f39]
        6       v8::Testing::DeoptimizeAll()  (in libchromiumcontent.dylib) + 187868  [0x105ed628c]
        6       v8::Testing::DeoptimizeAll()  (in libchromiumcontent.dylib) + 2914150  [0x10616fc16]
        6       v8::Testing::DeoptimizeAll()  (in libchromiumcontent.dylib) + 2915177  [0x106170019]
        5       uv__thread_start  (in Atom Framework) + 25  [0x1041fde65]
        5       v8::Testing::DeoptimizeAll()  (in libchromiumcontent.dylib) + 1252615  [0x105fda1b7]
        5       v8::Testing::DeoptimizeAll()  (in libchromiumcontent.dylib) + 1256537  [0x105fdb109]
        5       v8::Testing::DeoptimizeAll()  (in libchromiumcontent.dylib) + 1291152  [0x105fe3840]
        5       v8::Testing::DeoptimizeAll()  (in libchromiumcontent.dylib) + 1507080  [0x1060183b8]
        5       v8::Testing::DeoptimizeAll()  (in libchromiumcontent.dylib) + 1537770  [0x10601fb9a]
        5       v8::Testing::DeoptimizeAll()  (in libchromiumcontent.dylib) + 1594028  [0x10602d75c]
        5       v8::Testing::DeoptimizeAll()  (in libchromiumcontent.dylib) + 2747092  [0x106146f84]
        5       v8::Testing::DeoptimizeAll()  (in libchromiumcontent.dylib) + 2895758  [0x10616b43e]
```

**Comment by truevektordev** [comment]
I guess that isn't that helpful considering I don't have information on what it's trying to deoptimize...

**Comment by methodbox** [comment]
Just to be sure you guys are actually positive you don't have a ```.git ``` in your home PATH do this:

```$ls -a```
If you see a hidden git folder do this:
```$rm -rf . git```

It seems odd I could resolve this on a vanilla install and others can't.

Just to be sure I've noted it I have a 2014 rMBP i7 16GB with Nvidia 750m 512GB SSD.

**Comment by fbender** [comment]
> If you see a hidden git folder do this:
> $rm -rf . git

DO NOT do this if you want to keep your HOME folder!!

(There's an surplus space between the dot and `git`.)

**Comment by methodbox** [comment]
SORRY¡¡¡

Stupid iPhone.

Yes it should be ```$rm -rf .git```

**Comment by methodbox** [comment]
```#terminalisdangerous```

**Comment by methodbox** [comment]
#terminalisdangerous

**Comment by kohbo** [comment]
![image](https://cloud.githubusercontent.com/assets/6541769/8678619/7756779a-2a23-11e5-9020-3798f5ffed75.png)

I'm still having the same issue on multiple Windows boxes. This is while only having one 17KB html document open. I can't even use Atom because it hangs every few seconds.

**Comment by methodbox** [comment]
@kohbo A separate issue should be opened for this (or your issue should be reported in an existing issue for your OS) as it is quit possibly OS-specific.

All of the previous posts on this issue are related quite specifically to rMBP running OSX.

**Comment by kohbo** [comment]
@methodbox Alright, will do.

**Comment by mydearxym** [comment]
(mac 10.10.2) same issue both with atom or atom --safe, CPU extreme high ..

**Comment by truevektordev** [comment]
Somebody did something as of version 1.0.2. I'm not seeing high CPU usage anymore.

**Comment by mdalpi** [comment]
Just throwing a few cents here. Digging in my JS CPU profile I got:
![screen shot 2015-08-05 at 20 34 19](https://cloud.githubusercontent.com/assets/6289699/9100546/880ab92a-3bb2-11e5-95cd-d9eccc72af79.png)
As you can see from the bottom rectangle, 13% of my CPU is dedicated to update the vertical scrollbar. So checking [here (scrollbar-component)](https://github.com/atom/atom/blob/master/src/scrollbar-component.coffee) we can see that the scroll event is added, and when a scroll is detected the properties are updated for every-single-pixel (right?). 

I'm not an expert, and I really don't know if my theory is right, but this can explain why the Retina MacBooks are suffering more (the profile above is from my NON Retina 2010).

In the top rectangle we can see the exclamation points, and hovering it I get: *"Not Optimized: ForInStatement is not fast case"*. From my research, this is a V8 code optimization, maybe there's something that can be done, but I'll leave this with those who understand what they're talking about.

**Comment by chmac** [comment]
I found a potential workaround by force quitting the offending process in Activity Monitor and then telling atom to reload when it detects the crash. Has put my CPU load from >80% to <20%, and the editor continues to work as normal.

**Comment by zyklus** [comment]
Didn't profile this because I killed the process, but +1.  The obvious drop in CPU is when I killed "Atom Helper", got a message in Atom saying the editor crashed, then re-launched it and it was back to normal.

<img width="182" alt="screen shot 2015-09-02 at 12 11 39 pm" src="https://cloud.githubusercontent.com/assets/320562/9622912/ebf07398-516b-11e5-9c8c-093a5913de09.png">

**Comment by LegNeato** [comment]
Was told to repaste comment from https://github.com/atom/atom/issues/8914 here:

Atom is basically unusable for me because Atom Helper uses 100% CPU and the whole editor is unresponsive. Even exiting Atom tells me the editor is unresponsive and asks if I want to wait or close.

Reproduces 100% for me and with `atom --safe` as well. Also tried blowing away ~/.atom and it didn't fix it. Core i7 iMac, 10.10.5. Latest released Atom.

Here is the .cpuprofile from developer tools:

  http://pastebin.com/sPx4y5TM

To my untrained eye it looks like it is trying to download something and is in some tight busy-loop trying to emit status events.

And here is the sample of Atom Helper:

  http://pastebin.com/MpWYNd7G

Happy to do anything I can to help debug.

**Comment by chmac** [comment]
@LegNeato Did you try killing the Atom Helper? That resolved the issue for me, at least for the rest of that Atom run. I get a pop up asking if I want to restart it, and when I do, CPU usage is normal.

**Comment by LegNeato** [comment]
@chmac: Killing it for me makes the whole editor window no longer work. Selecting things from the menu (like Show Developer tools) also does nothing.

**Comment by LegNeato** [comment]
Interestingly, when Atom is locked up / unresponsive (with Atom Helper running), simply toggling the developer tools on via the menu makes Atom Helper go away and makes Atom responsive.

**Comment by LegNeato** [comment]
Looks like that was just a coincidence. Please let me know what I can do to help debug!

**Comment by mbakirov** [comment]
@LegNeato do you use http://nuclide.io? I've had same problem until deleted this package

**Comment by LegNeato** [comment]
Yeah, I used to manage the Nuclide team at Facebook ;-).

I think I see what is going on :crystal_ball: .I ran Atom Helper under `dtruss` :microscope:.

I  see that is is spidering every package and all their `node_modules` (to transpile?). When it gets to Nuclide, it has lots of packages with many deps and each of them are many layers deep such as:

`node_modules/nuclide-client/node_modules/nuclide-service-parser/node_modules/nuclide-node-transpiler/node_modules/babel-runtime/core-js/regexp/escape.js`

and because the spidering is single-threaded (I guess? I see see a bunch of mutex actions between each file read) they appear to effectively be transpiled serially and the process sits there at 100% CPU. I let Atom Helper sit there for for 30 minutes and eventually it finished and Atom was usable (Atom Helper went to 0% CPU).

I see some issues here:

1. The transpiling should be multi-threaded.
2. Even with `atom --safe` something (Atom?) is traversing 3rd-party packages and their node modules.
3. When the traversing / transpiling eventually finished, on next start Atom(?) does the exact same thing again.

Should I file these as separate issues?

Workaround:

* Entirely delete Nuclide packages off disk :cry: 

I also have the whole `dtruss` output I can send if someone is interested in seeing it.

**Comment by LegNeato** [comment]
For #3, it looks like `compiler-cache` is never written for js files (though it is for `cson` and `coffee`), even after I let the action finish. /cc @maxbrunsfeld

**Comment by greghe** [comment]
I have the same issue. 100%+ CPU. 1+ GB RAM. Locked up app.

**Comment by LegNeato** [comment]
Ok, did a bit of debugging. This is hard to chase down. I **think** that my problem is in `BufferedReader`...it seems to be in a tight read loop. I can comment out https://github.com/atom/atom/blob/8b40c3aaeba4c98730b8d5451b6aba1f09b55911/src/buffered-process.coffee#L75 and the CPu use goes away 100% after it was being repro'd 100%.

Interestingly, when I dump out the command `Atom Helper` is hung on:

 `/Atom.app/Contents/Resources/app/apm/bin/apm ls --json --no-color`

But I can't seem to find where someone calls that (atom or 3rd party package)!

**Comment by LegNeato** [comment]
Ok, my problem is definitely caused by the built-in `settings view` somehow. Master "works" because there is a coffeescript error, but when I load the latest release (https://github.com/atom/settings-view/commit/cb09a2d80fcab4c91b3cf47e7d6fe02174a4b1c0) I get the high CPU use.

When I comment out https://github.com/atom/settings-view/blob/master/lib/package-manager.coffee#L43, everything (except the settings view) works, no 100% CPU usage. I'll follow up on https://github.com/atom/settings-view/issues/555.

**Comment by LegNeato** [comment]
Found this while investigating FWIW:

https://github.com/atom/settings-view/pull/673

**Comment by LegNeato** [comment]
This code is in play on launch because of the status bar integration.

**Comment by LegNeato** [comment]
This completely fixed it for me :tada: :

https://github.com/atom/settings-view/pull/674

:8ball: Please test that it works for you too and report back! Here are the directions for Mac OS X:

1. Somewhere on your computer, `git clone https://github.com/LegNeato/settings-view.git`
2. `apm install settings-view`
3. `cd settings-view && apm link -d`
4. Then open Atom in dev mode `atom --dev`

Other platforms are probably on https://atom.io/docs/v0.186.0/contributing.

You should see significantly lower CPU. You can run fine in dev mode until the real fix is released (assuming it works).

**Comment by herrernst** [comment]
Another +1 from me. I have only one community package installed (ternjs). After some hours, Atom helper starts consuming around 100% CPU. If I close Atom, Atom Helper is still around and burns CPU, I need to force-kill it.
Here's the cpu sample: http://pastebin.com/Vh3vN54J
If I have time I try LegNeato's patch.

**Comment by nicbet** [comment]
On my 2014 rMBP high CPU occurs when a file is changed and a file changed indicator is visible (tabs / tree view packages - if disable these Atom behaves fine and sits at 1% CPU). 

Any CSS3 animation causes the CPU to spike. For example:

```css
-webkit-animation-name: pulse;
```

as used in brogrammer-ui theme.

**Comment by methodbox** [comment]
So at this point this issue has been open for a year, we've narrowed it down to what appears to be which ever Atom service/helper monitors file changes (note last comment on CSS animations and previous comments about a .git in your home/root folder), and at the very least isolated that there is indeed an issue affecting the laptop of choice of many (a majority?) of devs. Yet, I still see no action from the maintainers of Atom to address this.
I have to say, this doesn't seem to bode well for those of us looking to replace Sublime or our other favorite text editor.

It's not that I don't want to support Atom, I do and I'm excited to see it become something I can use everyday for coding, but right now it's not.

I think (again, as I have said it before) it's far too soon to call this a v1.

I don't have time to screw around with forcing my editor to work just so I can DO work.

**Comment by methodbox** [comment]
Also curious if anyone has tested this on So Capitan.

I haven't bothered opening Atom any time recently.

**Comment by nicbet** [comment]
@methodbox For what it's worth, I was able to resolve high CPU problem on my rMBP 13" (running El Capitan, by the way) by disabling "-webkit-animation" styles throughout. Atom.io has been my daily driver at work ever since and has now fully replaced Textmate.

**Comment by LegNeato** [comment]
v1.1 is out. Can everyone update and see if your issues are fixed?

**Comment by chris-aeviator** [comment]
still 100%+ cpu usage on atom helper :( 
(atom 1.1, yosemite, mbp late 2011)

**Comment by mydearxym** [comment]
still 100% cpu  --> atom 1.1  macOS 10.11 。。

**Comment by methodbox** [comment]
@chris-aeviator & @mydearxym - Can you see if your user home folder has a .git folder?

This was something we found previously contributed to this issue, and it's very common to have this folder.

See if removing this resolves the high CPU. Note that you'll need to have enabled viewing of hidden folders via terminal to see it in Finder, or run ```ls -al ``` in terminal while in your home folder.

I'm curious if this still persists.

**Comment by chris-aeviator** [comment]
@methodbox wow...this resolved the issue! I did not know about that folder, and it was almost 500 mb in size.

Atom now consumes around 5-25 % when idling.

Thanks for that, switching back to atom again :+1:

**Comment by methodbox** [comment]
No problem.  I'm not sure why but Atom scans this folder, which usually contains a lot of commits if you first setup git in your home folder.

**Comment by paambaati** [comment]
Well I don't have a `.git` folder in my home directory, but I still face the excessive battery drain. FWIW, I'm using Atom 1.1.0 on a late-2014 13" rMBP with 8 GB of RAM.

**Comment by chris-aeviator** [comment]
FYI:
removing .git folder also resolved high energy consumtion.

Before updating to Atom 1.1.0 i also applied the fixe indicated here (which did not fix). I don't know if this fix was merged into 1.1.0, maybe this helps to solve your issue @paambaati

**Comment by z-ax** [comment]
Yes, also decided to check the latest (v1.1.0) and confused by the issue.
But I have `nuclide-installer`, actually it was the reason to get the IDE for me. And don't have `.git` directory in my Home.

Issue happens when I open a mid-size project with Git, JS and some Java sources. And repeats when I open Settings Packages and Updates.

It's a pity, because I loved the functionality and current state in this direction.

MacBook Pro (Retina, 13-inch, Mid 2014).

**Comment by gbahamondez** [comment]
The same here, high and long CPU usage when updating packages  (8 packages at same time).

<img width="837" alt="screen shot 2015-11-06 at 1 17 09 am" src="https://cloud.githubusercontent.com/assets/3533121/10989289/2a7ce5a2-8424-11e5-8109-eacb8851a7f0.png">

**Comment by anguila** [comment]
Same thing. This is completely unusable... Is it even working for someone?

**Comment by FloFra** [comment]
To keep this issue clean, please do the following before reporting here:

Start Atom in safe mode i.e. with ```atom --safe``` and see if that fixes the issue:

 * If it does, a plugin is the root of the problem. Run Atom normally and disable the plugins one by one till the CPU usage drops. Once you found the plugin causing the issue, report your problem to the plugin developer.

 * If not, it is an Atom issue and this ticket is actually the right place to report the issue.

**Comment by methodbox** [comment]
@FloFra - You do realize this is still occurring with vanilla installations, right?

Not to say a plugin couldn't do the same, but I find it hard to believe github themselves couldn't easily reproduce the issue.

The fact that simply adding an empty .git repo to the home folder will cause the app to go apeshit should be evidence enough.

Since this problem began there's been no less than 3 active threads with no real response from the Atom owners acknowledging the issue, yet it is so easily duplicated.

Again, I think it's time to either own up that there is a serious problem or demote Atom from v1 to 0.9.

It's certainly not ready to be out of beta.

**Comment by benogle** [comment]
> Since this problem began there's been no less than 3 active threads with no real response from the Atom owners acknowledging the issue, yet it is so easily duplicated.

The hard thing is that high-CPU behavior can be caused by many many different things. e.g. https://github.com/atom/settings-view/pull/674 fixed one case, and we've fixed other cases (https://github.com/atom/atom/pull/8009, https://github.com/atom/atom/pull/5775) of similar behavior. Part of the team is currently focused on performance and is actively working on speedups in several dimensions and improvements to memory consumption. 

Issues are much easier for us to fix when they are discrete with clear reproduction steps. It seems there is an issue with a `.git` directory in the home folder. I too am not sure why it's reading that folder, but a separate issue detailing the problem with steps to reproduce would help us in understanding and fixing it. Please open that issue, and we can have a further discussion about it.

**Comment by FloFra** [comment]
@methodbox: Yes, that is excactly why I am trying to find a few people on the latest version of Atom that have the issue, so they can give us some reliable info and we can find out what is the root of the problem. 

Right now what we have here is not really a bug report but a bunch of people complaining about performance issues. This ticket once started to complain about idle usage of 10% and now contains a bunch of reports about AtomHelpers being stuck at 100%, indicating a much different and more grave problem.

Keep in mind that if you want the issue fixed, we need to pin it down to a cause first. Trying to reproduce it, is the most important first step.

On a last note more personal note: Nobody is forcing you to use this free and open software. If it is not working for you, I am sorry. But coming on here attacking people that are trying to help when you, as you stated yourself, have not even "bothered to open Atom any time recently" is probably not to way to fix that.

**Comment by methodbox** [comment]
Ok let me be clear on summarizing the issue, as it's been duplicated both here and the Atom forums:

Install El Capitan or Yosemite on any supported machine.

Install Atom.

Run Atom.

It's literally that easy to reproduce, to point that I've actually done it two different machines, fresh installations.

I just find it hard to believe with the number of other users experiencing the same thing that you need examples beyond what's been provided.

Yes, I could use other software for coding, and I do. I still use Sublime 2 because it works just fine on El Capitan and barely consumes any resources.

I haven't been able to use Atom as a viable coding platform since its release, in any form.

You are correct, it's free and my choice to use. But your users have every right to be skeptical based on the responses received.

**Comment by 50Wliu** [comment]
> It's literally that easy to reproduce, to point that I've actually done it two different machines, fresh installations.

None of the core Atom team have been able to reproduce this, so it's definitely not as simple as that.

Examples are needed because not everyone is facing the same bug, as @benogle pointed out.

**Comment by zachwood** [comment]
I had the same problem a few weeks ago, then uninstalled this ternjs package and everything is fine now.

https://atom.io/packages/atom-ternjs

**Comment by bgnlordhelmut** [comment]
As a way to help debug this, I removed literally as much as I could find related to Atom on my system, essentially uninstalling it:

~/.atom
/usr/local/bin/atom
/usr/local/bin/apm
/Applications/Atom.app
~/Library/Preferences/com.github.atom.plist
~/Library/Application Support/com.github.atom.ShipIt
~/Library/Application Support/Atom
~/Library/Saved Application State/com.github.atom.savedState
~/Library/Caches/com.github.atom
~/Library/Caches/Atom

I then downloaded latest (v.1.2.3), installed, and it's running fine again with no memory issues. I haven't installed any plugins yet, so we'll see if anything in particular causes it to resume possible leaks/CPU issues. If I find any, I'll report them here.

**Comment by gscottolson** [comment]
Atom has been running wild on my MacBook Pro. Intermittent responsiveness issues and many force quits. This morning, I ran `atom --safe` and it is clearly helping. Haven’t seen the CPU usage above 5%.

**Comment by Zireael07** [comment]
@gscottolson: This means it's a plugin issue.

**Comment by nicolazj** [comment]
replace /Applications/Atom.app/Contents/Resources/app/apm/bin/node with newer version, like 0.12.*

It works perfectly now

**Comment by lee-dohm** [comment]
As I mentioned in [my post on Discuss](https://discuss.atom.io/t/atomhelper-high-cpu-usage/2553/111?u=leedohm), replacing that copy of node doesn't change the version of node that Atom is running on.

**Comment by houmark** [comment]
For me it looks like the pigments package is making Atom Helper going crazy and crazy CPU. Disabling it and I'm back to normal CPU.

**Comment by esetnik** [comment]
@houmark thanks for the tip. I had the exact same issue and disabling the pigments package brought my cpu usage on Atom Helper from 100% to 0%

**Comment by houmark** [comment]
@esetnik Yeah, it seems like this otherwise great package has some performance issues in some situations.

@abe33 For your reference. Not sure how I can be helpful with sending you debugging information about my performance issues - would love to get pigments back in the arsenal :)

**Comment by methodbox** [comment]
Is pigments one of the default packages?

If you guys need feedback, a clear example of how this should be provided, which error logs, commands we can try would be useful.

Also, running in safe mode isn't helpful. It may resolve the issue temporarily but it doesn't help us reproduce and narrow the cause.

**Comment by houmark** [comment]
No, it's not, but it's featured and basically shows up as the first featured package (for me at least), so as close to a core package as can be. Close to 300k downloads...

**Comment by esetnik** [comment]
I think this issue filed against atom can be closed unless someone can reproduce the high cpu usage in safe mode. Otherwise it's going to turn into a never-ending list of new incidences of high cpu usage. As it is now this comes up first in searches for "Atom Helper 100 CPU". The large majority of the performance issues i've encountered are related to third-party plugins and should be filed with the package maintainers. Thanks for the pigment tip!

**Comment by wallverb** [comment]
I can reproduce it constantly even when running in safe mode. The way to reproduce is the following (at least in my case):
1. I have files in my project: `logic/tools.js` and `routes/tools.js`
2. I have single pane with `logic/tools.js` open
3. Click `View / Panes / Split Right`
4. Now you have `logic/tools.js` open in both panes
5. Hit `Cmd + P` and navigate to `routes/tools.js` 
6. Atom freezes, Atom Helper gets to 100% and I need to kill the Atom

It happens both with `atom` and `atom --safe`

**Comment by 50Wliu** [comment]
Hi @wallverb, the bug you're describing has already been fixed in the Atom 1.4 betas.  Sorry for the inconvenience.

**Comment by actsasgeek** [comment]
I just downloaded Atom and am running into the same problem. I am using Atom 1.3.2 on a brand new rMBP (mid-2015) using OS X 10.11.1. I honestly don't know what else to tell you except that **I didn't even have time to install 3rd party packages**. AtomHelper is running at 85-100% and when I quit Atom, it doesn't always die.

This happens whether I click on the Atom icon, use `$ atom` or `$ atom --safe`. About 7-10 seconds after the window opens, the Atom Helper appears in Activity Monitor at around 85-100% CPU usage.

**Comment by methodbox** [comment]
@esetnick I am not sure you've read the full topic if you can't reproduce these results.

As the last post said - exactly as mine said months ago - this happens on a brand new install of OS X with a brand new install (hence no 3rd party involved) of Atom.

Are you suggesting for Atom to function correctly we need to always run in safe mode?

There's clearly still something wrong with the core of Atom.

I've done a clean OS X install on the last three versions and the problem persists.

**Comment by actsasgeek** [comment]
I think I solved part of the mystery--at least for myself.

Undeterred by a single pegged CPU, I kept on using Atom. At one point, I cloned a repository from the internets and started browsing the files. Having gotten used to my fan running all the time, I quickly noticed that it was *not* running Weird but I remembered that the two projects I'd looked at earlier where brand new and *not* under source control. This suggested to me that the problem was related to source control.

I went back this morning and loaded one of those new projects into Atom. Yup, 10 seconds later, AtomHelper is pegged to 100%. Close everything, force quit the AtomHelper process, run `git init` on that folder and reload everything? AtomHelper is at 1%.

One more piece to the puzzle. My prompt is set to show the git branch I'm on. One of the strange things I noticed when I set up this computer was that I was *always* on a branch. `cd ~/Desktop`, on master. `cd ~/Documents/workspace`, on master. I could never quite figure out why that was the case.

I reasoned that the Git integration code in Atom probably works by recursively searching for a `.git` folder and only stops when it reaches `/` so I started looking for a wayward `.git` directory. When I opened up `~`, I found one. When I deleted it, AtomHelper stopped misbehaving on folders that were not under source control.

What I assume was happening is that having found a `.git` directory in my home directory, Atom was trying to analyze my entire home directory. This seems like a misfeature. If I open up a directory no matter how far up in the hierarchy Atom has to search for the `.git` directory, it should only be analyzing the directory I opened.

You should be able to reproduce *my* problem by running `git init` in your home directory and then opening a directory/project 2 or 3 levels down in your Documents directory (on OS X).

**Comment by 50Wliu** [comment]
@actsasgeek Thanks for taking the time to look into this :smile:.  It sounds like you're experiencing the same root cause as #6437.  Can you please try to take a CPU profile and upload the results here?  If you don't know how, [these instructions](https://atom.io/docs/latest/hacking-atom-debugging#diagnose-performance-problems-with-the-dev-tools-cpu-profiler) should help you.

**Comment by methodbox** [comment]
@actsasgeek
Yes, as @50Wliu said, this issue has been known for some time, though your description makes it more clear that it's likely related to Atom's strange need to index everything even remotely related to any .git folder (still not clear why this is even a "feature").

That said, it still doesn't explain the seemingly identical behaviour in a fresh OS X install with a fresh Atom install which does the same... Or does it?

If Atom is recursively searching for a .git until reaching root ( / ) then doesn't find this, then what?

1) It shouldn't do this. The expected behaviour is as described by Actasgeek.

2) If it is searching for a .git and not finding one, my guess is there's no if{} stopping it from continuing forever until it finds one.

3) If it doesn't find one yet continues searching recursively for one, this is effectively a permissioned endless loop.

It's been known for a while that simply having a .git in your home directory would instantly cause this issue. Now it seems we might know why.

**Comment by joshaber** [comment]
For those of you running into the git-in-my-home-directory problem, it's been resolved in ##10249.

**Comment by methodbox** [comment]
Nice. I'll try a new install and see if #10249 resolves my issue.

**Comment by edocz** [comment]
I just clean ~/.atom/packages and building nuclide from Source at /tmp and everything is fine!

**Comment by FyodorPavlovich** [comment]
@joshaber I'm not sure that it has been. On Friday I obtained version 1.4.1, but still found Atom chewing through 1Gb+ of ram. I did find that deleting the .git folder from my home directory completely resolves the issue.

**Comment by 50Wliu** [comment]
@FyodorPavlovich The fix is only in Atom 1.5.0 (currently in beta) and above.  Sorry about the delay.

**Comment by FyodorPavlovich** [comment]
@50Wliu Oh I see, thanks.

**Comment by timaschew** [comment]
1.5.0 beta1 is still freezing on OS X, but not so long

**Comment by timaschew** [comment]
update: now it also freezes until a message appear and I need to restart atom

**Comment by MatrixV01** [comment]
I noticed that only when I tried to create/edit a new local file/folder that the Atom helper would destroy cpu usage. When I linked the same file/folder to a Github repo, it went back to normal. Not sure why helper goes crazy when you're in a file not associated with a repo.

**Comment by kexoth** [comment]
@MatrixV01 do you have any git-related plugins?
I had the same issue, I since I installed `git-diff`, `merge-conflicts`, `tree-view-git-status` etc. 
After I read your comment I disabled them to see if some of them cause this issue.

**Comment by methodbox** [comment]
Yeah just checking back, and I am also still experiencing no change in Atom performance on OS X.

To be clear, that is to say simply doing the following results in basically eating resources on my Mac:

1) Install Atom on El Capitan
2) Run Atom

Not trying to be a smartass, but if you guys can't reproduce this issue, then you're not testing the simple basics.

I have to question whether the supporting devs on this project are actually testing under a vanilla install.

My current Mac OS X doesn't have a .git in the home directory, because it's never existed- it's a vanilla OS X install.

Combine this issue with #9185 where users on a vanilla install are reporting freezing and I think it's clear that Atom on OS X is still experiencing beta-level issues.

I'm sorry but, again I have to wonder if you guys are trying to duplicate our issues on El Capitan or a previous version of OS X.

**Comment by TatumCreative** [comment]
I have some specific steps to reproduce on my Macbook running OS X El Capitan, 10.11.4 involving an excessively large repo.

* `git clone git@github.com:mozilla/gecko-dev.git`
* `atom --safe gecko-dev`

CPU usage goes crazy in AtomHelper, 99%. Sometimes when closing out of atom the processes stick around as well.

* `mv gecko-dev/.git gecko-dev/git`

CPU usage is down to sane levels in AtomHelper.

**Comment by methodbox** [comment]
@TatumCreative
Interesting, as this issue was in the beta for a while; if there was even a .git in your home directory it would cause this behavior.

However, that bug was closed a long time ago, the. A similar issue opened, and Atom devs said it was resolved.

When you cloned this repo, what path did it land in?

**Comment by jbskis** [comment]
I disabled the 'pigments' package and it dropped from 100 percent to 1 percent!!

**Comment by djwgen** [comment]
Based on @jbskis' post, I also disabled 'pigments' and saw the CPU usage of Atom Helper drop back to sane levels.

**Comment by TatumCreative** [comment]
@methodbox It does that at `~/desktop`, and I have no .git folders in my home directory.

I have 2 solutions so far that lower my CPU usage.

### Solution 1
I symlink the subfolder I'm working in to `~/my-subfolder` and then work there. That way atom doesn't know about version control and CPU usage is minimal.

### Solution 2
Use this [forked branch](https://github.com/jryans/atom/tree/ahead-behind) that comments out one of the more [intensive repo checking lines](https://github.com/jryans/atom/commit/f1def0cfd1d95e7d11ff646202a276c787506ced).

Run `script/build --verbose --install-dir ~/Applications/AtomDebug.app"` to build a custom build.

**Comment by electricarts** [comment]
The solution here is disabling the "Pigments" package. After disabling it the cpu usage goes back to 3%. Enabling the package the usage is 103%. Repeated it several times. Same result.

**Comment by gugote** [comment]
Solution for me was disabling the Pigments package too. Sadly, because I love that package, but I can confirm that it was the cause of the trouble.

**Comment by Benjamski** [comment]
Pigments package for me as well. I know I'm just echoing previous sentiments here but it really is a damn shame. I only encountered this issue today (just started using pigments today) but Atom was using 102% of my cpu and so sluggish it was entirely unusable.

**Comment by qolop** [comment]
+1. The Pigments package was the root of the issue for me on Ubuntu 16.04.

**Comment by sercanov** [comment]
I had no pigments package but disabling`linter-flow` solves it too. eslint seems enough for now ^^

**Comment by methodbox** [comment]
@lee-dohm I'm really curious about a post you made on #1827 (closed) and exactly who you mean is the actual target audience for Atom. It seems like you're suggesting this is an editor that isn't made for "the masses" to use on their "standard" machines:

> There are an infinite number of individual reasons why Atom might be consuming too much CPU. The general categories are:

Atom isn't designed to be a tiny native-code editor. If the ultimate in speed is what you're after, Atom probably isn't what you're looking for at least for now ... and possibly not ever depending on what you're expecting.

**Comment by past** [comment]
I do see this in 1.10 on latest OS X even with all non-core packages disabled. I don't think I saw that in 1.9 (at least not constantly), but I might be wrong.

**Comment by JohnMica** [comment]
Seems this issue is old but it is still present today. On macSierra GM the helper runs cpu @ 50-120% while atom itself is mostly unresponsive... :( have to keep waiting for a new release as I haven't touched the conf besides fontfamily Monaco ....

**Comment by lee-dohm** [comment]
There are an infinite number of individual reasons why Atom might be consuming too much CPU. The general categories are:

1. Atom isn't designed to be a tiny native-code editor. If the ultimate in speed is what you're after, Atom probably isn't what you're looking for at least for now ... and possibly not ever depending on what you're expecting.
1. An Atom package is causing a problem
1. Atom itself is causing a problem

If Item 1 is the root of the problem, there are many other text editors that will probably suit your needs. [Sublime Text](https://www.sublimetext.com) is a popular one. [vim](http://www.vim.org) is another. They were both designed with very different goals than Atom and if they suit your needs better, we're happy if you're happy :grinning:

If item 2 or 3 is more what is happening for you, then I invite you to take a look at the improved [Atom Flight Manual](http://flight-manual.atom.io). There is [a tutorial on diagnosing runtime performance issues](http://flight-manual.atom.io/hacking-atom/sections/debugging/#diagnose-runtime-performance) that will help you give us the information we need to track down these CPU problems.

So here's what I want people to do for us so that we can get these problems identified, root caused, and fixed:

1. Determine to the best of your ability what actions or situations cause the problem
    * Does the problem only occur in certain projects or in all projects?
    * Does the problem occur right away or only after a long period of time?
    * Does the problem only happen when you do something specific like "only when I use Find in Project and include a file mask"? Or is it more of a general thing like "any time I use find-and-replace"?
1. See if you can reproduce the problem in [Safe Mode](http://flight-manual.atom.io/hacking-atom/sections/debugging/#check-if-the-problem-shows-up-in-safe-mode). Whether or not you can reproduce the problem in Safe Mode, we want to help with the problem ... but we need to know where to start looking. Remember, the longest `git bisect` session starts with a single division of the problem space :grinning:
1. Use the [runtime performance diagnosis tutorial](http://flight-manual.atom.io/hacking-atom/sections/debugging/#diagnose-runtime-performance) to gather a CPU profile

Then, once you've done the above:

* If you **can reproduce the problem reliably**, open a **new Issue**
    1. Fill out the template
    1. Include what version of Atom you're running by including the output of `atom --version`
    1. Include what OS and version you're running Atom on
    1. Include as much information as you can about what actions or situations cause the problem
    1. Include a mention of #1827 in the new Issue so that we know you're following these instructions
    1. Include the CPU profile that you recorded in the new Issue
* If you **need help reproducing the problem**, go to [Discuss, the official Atom message board](https://discuss.atom.io) and ask for help. There are plenty of people there (including myself and Atom volunteers) that will do their best to help you figure out what is going wrong so you can file an Issue to help us get it fixed!

We want to get all of these bugs nailed down, but like I said in [my recent blog post](http://blog.atom.io/2016/04/19/managing-the-deluge-of-atom-issues.html), Issues need to be actionable. There needs to be a plan of action. This Issue doesn't have that, so I'm going to close it and ask people to work with me to help us tease out all the individual bugs from this mess.

Thanks for reading and I'm looking forward to seeing all the new Issues!
